### ISSUE TITLE
When creating the initial README.md from the project creation page, the text lines have CRLF line endings when the client is Windows, but Gitea is running in Linux

### ISSUE THREAD

**Issue by alexlehm** [go-gitea/gitea#27120]
### Description

When creating the project description README.md file, the resulting file in the git has CRLF line endings on the part the is copied from the project description, when checking out in Linux the file is still using CRLF endings since no autoconversion is used.

### Gitea Version

1.20.4

### Can you reproduce the bug on the Gitea demo site?

No

### Screenshots

when looking at the resulting file checked out on Linux, it looks like this:

![image](https://github.com/go-gitea/gitea/assets/156144/c74e246a-8c83-4ecf-ba6c-c0369bf9803b)

### Git Version

2.30.2

### Operating System

ubuntu  22.04.2 LTS

### How are you running Gitea?

GITEA_WORK_DIR=$PWD/data/ ./gitea web -c $PWD/etc/app.ini -p 3003

### Database

sqlite

**Comment by lng2020** [comment]
I can't reproduce this bug while submitting the README file in Windows and viewing the file in Linux. Do you have any specific details to reproduce?
Or you can check the git setting following this guide(https://stackoverflow.com/questions/35801024/how-to-see-what-type-of-line-endings-are-already-in-git-repository) and config them as described in(https://docs.github.com/en/get-started/getting-started-with-git/configuring-git-to-handle-line-endings?platform=windows)

**Comment by lng2020** [comment]
The current behavior of Gitea:
1. If you edit a file through the Gitea web editor, the CRLF will auto-translate into LF
2. If you upload a file through the web button, the CRLF will be unchanged.
3. If you upload a file through git, it depends on the config `core.autocrlf` and `.gitattributes`, see(https://docs.github.com/en/get-started/getting-started-with-git/configuring-git-to-handle-line-endings?platform=windows)

**Comment by alexlehm** [comment]
This only happens when typing the text in the description form field, this is copied into the README.md file along with the project name and the lines from the description end up with CRLF

I fixed that by doing recode -f on the projects I created, I wonder if it would be possible to filter the description to remove the CR chars before creating the file

**Comment by lng2020** [comment]
> This only happens when typing the text in the description form field, this is copied into the README.md file along with the project name and the lines from the description end up with CRLF
> 
> e.g. here. https://tildegit.org/alexlehm/test-test2
> 
> I fixed that by doing recode -f on the previous projects I created, I wonder if it would be possible to filter the description to remove the CR chars before creating the file
> 
> In fact it might work to open the README file with the web editor once and save it again to remove the CRs from the text

Well I *guess* the code forgets to remove the "\r" before generating the README.md file automatically. Will try to figure and fix this one later.

**Comment by alexlehm** [comment]
(the same issue is present in other forks of the software including gogs)

**Comment by silverwind** [comment]
How exactly is the README generated for the issue to show? It is through this option in repo creation?

<img width="587" alt="image" src="https://github.com/go-gitea/gitea/assets/115237/7275e0b7-41de-4eb3-81f5-c492399589c3">

**Comment by alexlehm** [comment]
I use Firefox in Windows and type multiple lines in the description field and create the default README.me file

****

it does not related to the web editor

**Comment by silverwind** [comment]
I see. Wasn't aware we copy the description field into the generated README.

**Comment by alexlehm** [comment]
the odd thing is, this feature is specific to gitea (and forks), both github and gitlab have a single-line description line that is copied into the readme which likely avoids the lineend issue

**Comment by silverwind** [comment]
So I guess we could normalize to LF here which is the most common EOL format in git repos overall. We have no .editorconfig or any other indication on which EOL preference the user has at this point, so we have to choose one format and LF seems the most likely correct one.

**Comment by alexlehm** [comment]
i thought that as well, any README file will be unix format, if the user uses Windows they will usually have autocrlf active in their git config for the client

**Comment by lng2020** [comment]
<img width="263" alt="image" src="https://github.com/go-gitea/gitea/assets/70063547/e916331f-d409-4e0c-8a9d-62e1315c08f0">

Currently, due to the PR(#27141), the Monaco editor can auto-detect the EOL. So it won't encounter an error message even if the EOL is CRLF.  That's the exactly same with Github. I think it's better than changing the content silently. 
Is it okay to close it now?

**Comment by lng2020** [comment]
>  I would normalize this to LF so that we don't end up with a mixed EOL file.

Indeed. The README.md above is a mixed EOL file but the Monaco editor just work well.
<img width="261" alt="image" src="https://github.com/go-gitea/gitea/assets/70063547/ebed5e09-db00-4776-b4e0-c152174a8dcf">

**Comment by silverwind** [comment]
Fixed by https://github.com/go-gitea/gitea/pull/27141.

**Comment by wxiaoguang** [comment]
> Fixed by [#27141](https://github.com/go-gitea/gitea/pull/27141).

Are you sure?

The author said clearly: https://github.com/go-gitea/gitea/issues/27120#issuecomment-1732269057

**it is not related to the web editor**

**Comment by silverwind** [comment]
Then this issue is between author and git only, not sure if Gitea can do something about it.

**Comment by wxiaoguang** [comment]
> Then this issue is between author and git only, not sure if Gitea can do something about it.

Spend some time on studying how the repo README is initialized.

**Comment by wxiaoguang** [comment]
The fix: Fix repo init README EOL #37388

```
"Description":    util.NormalizeStringEOL(repo.Description),
```
